### DISCUSSION TITLE
Publish Blazor WebAssembly as static files and folders

### DISCUSSION THREAD

**Discussion by ParsaLmn** [bitfoundation/bitplatform#5217] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Hi
I want to use admin template WebAssembly mode but i have my own server api project and I dont want to use
bit admin template server api project. 
so as I read in documentations for running WebAssembly mode the api project has to be set as the startup project.
and when I publish the project the size of it is so large which most of it is unuseful for me.

### Describe the solution you'd like

Is there any other way to publish on WebAssembly  mode without using api project?
Or may i create a simple api project just for running the client web project on WebAssembly mode?

**Comment by yasmoradi** [comment, marked as answer]
Hi there,

Thanks for reporting the issue.

We're currently using `_Layout.cshtml` and `asp-append-version=true` to make CSS and JS files http cache as efficient as possible, and also to make prerendering possible.

To address your issue, I recommend moving your existing API code to the project template's API project.
This is a simple API project that you can customize to your needs. By doing this, you can publish a single project.

Another option is to use your existing API's address in the `src/Client/Core/appsettings.json` file. This will require you to publish two projects, but you can reduce the size of the published output by removing unused NuGet packages, such as `HealtCheck`.

We do not currently have plans to support Blazor WebAssembly with an `index.html` file so that it can be published as a static website.
We might consider this for the future.

**Reply by ParsaLmn** [reply]
Hi.
i have added my API's address in the `src/Client/Core/appsettings.json` file as you mentioned.
and then i deployed the Admin Template on WebAssembly mode.

But now the problem is the project crashes after a few seconds.
I think the problem is with ServiceWorker  because the project stucks at `New version is available Downloading updates, please wait` and then it shows the `503 Service Unavailable` message.
